### ISSUE TITLE
Statement on HW / SW TPM

### ISSUE THREAD

**Issue by lukehinds** [keylime/keylime#591]
I think we could be a little less forceful in this view

> Keylime can be used with a hardware TPM, or a software TPM emulator for development, testing, or demonstration purposes. However, DO NOT USE keylime in production with a TPM emulator! A software TPM emulator does not provide a hardware root of trust and dramatically lowers the security benefits of using keylime.

A swtpm is for sure does not provide the same depth of guarantees as a HW tpm, however its still superior to something like AIDE. Someone could use a cloud providers swtpm (google cloud provide them). Granted, they are trusting the providers infra then, but that is better than solely trusting the host. 

I recommend we remove the 'DO NOT USE' and instead explain the situation in that the are different levels of guarantee:

1. HW, you trust the chip
2. SWTPM, you trust the underlying host
3. SWTPM spawned soley in the node, you're trusting the same node as which you're measuring

I under 3 is poorly worded and would need some work.

**Comment by lukehinds** [comment]
tagging tpm folks @stefanberger @puiterwijk

**Comment by stefanberger** [comment]
Yes, for software TPM you have to trust the underlying host and admin(s).

**Comment by mpeters** [comment]
I agree with the spirit of this change. "DO NOT USE" is too harsh. It's just different tradeoffs as it does provide protection against a lot of intrusion attacks, just not a malicious hypervisor.
